### ISSUE TITLE
Crashed with SIGSEGV in check_cursor_lnum

### ISSUE THREAD

**Issue by Neur1n** [neovim/neovim#9506]
- `nvim --version`: NVIM v0.4.0-dev (installed using unstable ppa, also happened in v0.3.2)
- Vim (version: ) not installed
- Operating system/version: Ubuntu 16.04
- Terminal name/version: bash
- `$TERM`: xterm-256color

### Steps to reproduce

This issue happens with the following minimal `init.vim`.
```
call plug#begin('~/.config/nvim/plugged')
Plug 'wesQ3/vim-windowswap'
call plug#end()
```

### Actual behaviour

When entering any buffer, or just an empty one for example, using `:h help` (or any other) and the doc buffer split opened. Then use `<leader>ww` twice to swap the empty buffer and the doc buffer. Now the `concealevel` has been changed. One thing to be noticed is that, before swapping, using `:verbose set conceallevel` in the empty buffer (let's tag it with B1) gives out `conceallevel=0` (M1), and in the doc buffer (B2)  give out `conceallevel=0 Last set from /usr/share/nvim/runtime/ftplugin/help.vim` (M2) which means B1->M1 and B2->M2. Then after swapping B1 and B2, it becomes B2->M1, B1->M2 which seems the settings are not following the layout change. And now if I quit nvim from the doc buffer, nvim crashes.  But if swapping by using `<C-w>j/k` this will not happen. I wish I explained it well... (Related [issue](https://github.com/wesQ3/vim-windowswap/issues/14))

![1](https://user-images.githubusercontent.com/17579247/51153708-2bcff000-18ac-11e9-86ce-512dfe37b325.png)

![2](https://user-images.githubusercontent.com/17579247/51153716-35595800-18ac-11e9-94e3-d032aa006e73.png)

![3](https://user-images.githubusercontent.com/17579247/51153725-3c806600-18ac-11e9-9478-cf7e2bd17fbd.png)

![0](https://user-images.githubusercontent.com/17579247/51154059-8453bd00-18ad-11e9-8242-cb16710cd393.png)

### Expected behaviour

Nvim does not crash.

**Comment by grifcj** [comment]
Could not reproduce with either NVIM v0.4.0-753-gfbf2c414a or the app image of v3.2. Running on Ubuntu 19.04.

Followed directions on linked issue on plugin page; found them a bit clearer. I noticed that the conceal dropped to 0 for help window, but when I quit it, it just quits without crash. Leaves me with empty buffer, which I can quit to completely quit nvim.

Wanted to create a gif for it, so you could see my steps exactly, but was having some troubles there.

**Comment by justinmk** [comment]
duplicate of https://github.com/neovim/neovim/issues/9773, which has a fix in 0.4
